examples/async_fl/resource_manager.py

```diff
@@ -12,17 +12,15 @@ def __init__(self, experiment_mode):
         self.experiment_mode = experiment_mode
         self.update_lock = threading.Lock()
 
-    def get_remaining(self):
-        return len(self.client_run_queue)
+    def get_task_length(self):
+        self.update_lock.acquire()
+        remaining_task_num: int = len(self.client_run_queue)
+        self.update_lock.release()
+        return remaining_task_num
 
     def register_tasks(self, clientsToRun):
```

Comment:
Do the tasks need resorting based on end time when new tasks are registered, or is there some assumption that makes this unnecessary? For example, let's say `aggregation_buffer_size=1`, in round 1 we have sample client_A[start=1, end= 3] and client_B [start=2,end=10]; in round 2 we sample client_C[start=5,end=8]. In round 2, we would actually get the result of client_B and aggregate it before client_C because it is earlier in the queue, even though client_C should have finished earlier. 

Also, with multiple executors, the tasks might return in a different order from the original queue here. How is this handled?

Reply:
These are pretty good points. 

1. We can mitigate the first concern by prefetching more client selection in future rounds ([here](https://github.com/SymbioticLab/FedScale/pull/162/files#diff-ceaf64e0eb8e76f1f34f2d04c7ba134994157ccca358e9b3b56d9920d76596c4R189)). 
2. The second problem leads to a tradeoff between simulation efficiency and fidelity. One workaround is to stall get_next_task before clients of the current buffer_size complete;

Please let us know your feedback. Thanks!

Reply:
1. That makes sense, thanks for clarifying. I think it's a valid workaround. My only concern is that with "prefetching", the difference is that round number will affect aggregation because of the staleness factor. Btw, is there a mechanism to keep the size of this queue from exploding since we are preselecting a lot? 
2. I think that's a good solution / option to have for correctness. Simulation speed is not a huge concern compared to the fidelity of the results. If one worker is somehow running slower than the others, then at least one result would be out of order each round, which could potentially change the results. 

Thanks a lot for addressing my feedback so quickly!

Reply:
My pleasure! 

1. I am not pretty sure whether I have understood your concern correctly, please feel free to correct me if not. I think "prefetching" is fine, since we are tracking their correct "(completion) round number", thus staleness, after sorting by the completion time. Please note that the selection_round is different from the completion_round we used in aggregation. In fact, I guess we even do not care about the "selection_round", if we ignore its impact on participant_selection decisions? Admittedly, we are sacrificing the selection fidelity, but it is hard to circumvent this further in simulations.
2. Sounds good. I will fix this soon.

Thanks a lot for your input! :)